applications/MappingApplication/custom_mappers/nearest_neighbor_mapper_iga.h

```diff
@@ -0,0 +1,294 @@
+    {
+        rValue = mNumberOfNearestNeighbors;
+    }
+
+private:
+    std::vector<double> mShapeFunctionValues;
+    std::vector<int> mNearestNeighborId;
+    IndexType mNumberOfNearestNeighbors;
+    double mNearestNeighborDistance;
+
+    friend class Serializer;
+
+    void save(Serializer& rSerializer) const override
+    {
+        KRATOS_SERIALIZE_SAVE_BASE_CLASS( rSerializer, MapperInterfaceInfo );
+        rSerializer.save("ShapeFunctionValues", mShapeFunctionValues);
+        rSerializer.save("NearestNeighborId", mNearestNeighborId);
+        rSerializer.save("NumberofNearestNeighbors", mNumberOfNearestNeighbors);
+        rSerializer.save("NearestNeighborDistance", mNearestNeighborDistance);
+    }
+
+    void load(Serializer& rSerializer) override
+    {
+        KRATOS_SERIALIZE_LOAD_BASE_CLASS( rSerializer, MapperInterfaceInfo );
+        rSerializer.load("ShapeFunctionValues", mShapeFunctionValues);
+        rSerializer.load("NearestNeighborId", mNearestNeighborId);
+        rSerializer.load("NumberofNearestNeighbors", mNumberOfNearestNeighbors);
+        rSerializer.load("NearestNeighborDistance", mNearestNeighborDistance);
+    }
+};
+
+class KRATOS_API(MAPPING_APPLICATION) NearestNeighborLocalSystemIGA : public MapperLocalSystem
+{
+public:
+
+    explicit NearestNeighborLocalSystemIGA(NodePointerType pNode) : mpNode(pNode) {}
+
+    void CalculateAll(MatrixType& rLocalMappingMatrix,
+                      EquationIdVectorType& rOriginIds,
+                      EquationIdVectorType& rDestinationIds,
+                      MapperLocalSystem::PairingStatus& rPairingStatus) const override;
+
+    CoordinatesArrayType& Coordinates() const override
+    {
+        KRATOS_DEBUG_ERROR_IF_NOT(mpNode) << "Members are not initialized!" << std::endl;
+        return mpNode->Coordinates();
+    }
+
+    MapperLocalSystemUniquePointer Create(NodePointerType pNode) const override
+    {
+        return Kratos::make_unique<NearestNeighborLocalSystemIGA>(pNode);
+    }
+
+    void PairingInfo(std::ostream& rOStream, const int EchoLevel) const override;
+
+    void SetPairingStatusForPrinting() override;
+
+private:
+    NodePointerType mpNode;
+
+};
+
+/// Nearest Neighbor Mapper for IGA-FEM simulations 
+/** This class implements the Nearest Neighbor Mapping technique for IGA-FEM partitioned simulations.
+* Each node on the destination side (FEM side) gets assigned is's closest integration point (neighbor) on the other side of the interface (IGA side).
+* Once the nearest integration point is found, the shape function values of the integration point are used to interpolate the values from the IGA side to the FEM side.
+* For information about the available echo_levels and the JSON default-parameters
+* look into the class description of the MapperCommunicator
+*/
+template<class TSparseSpace, class TDenseSpace, class TMapperBackend>
+class KRATOS_API(MAPPING_APPLICATION) NearestNeighborMapperIGA
```

Comment:
you need this only because you need to customize the implementation via `NearestNeighborInterfaceInfoIGA` and `NearestNeighborLocalSystemIGA` right? Asking since for the NE mapper it was not necessary, maybe the base NN classes could be changed to accomodate IGA, but not sure